Week5/app/src/main/java/com/isabelledionisius/mymovieapp/MovieViewModel.kt

```diff
@@ -0,0 +1,36 @@
+package com.isabelledionisius.mymovieapp
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+
+class MovieViewModel(application: Application) : AndroidViewModel(application) {
+
+    private val repository: MovieRepository
+
+    // Using LiveData and caching what getAlphabetizedWords returns has several benefits:
+    // - We can put an observer on the data (instead of polling for changes) and only update the
+    //   the UI when the data actually changes.
+    // - Repository is completely separated from the UI through the ViewModel.
+    val movieItems: LiveData<List<MovieDataList>>
+
+    init {
+        val movieDao = MovieRoomDatabase.getDatabase(application).wordDao()
+        repository = MovieRepository(movieDao)
+        movieItems = movieDao.getMovieItems()
+    }
+
+    /**
+     * Launching a new coroutine to insert the data in a non-blocking way
+     */
+    fun insert(movieItems: MovieDataList) {
+        GlobalScope.launch (Dispatchers.Main) {
```

Comment:
I'm not sure if this is Week 6 or Week 5 that you've added it, but the Homework **explicitly** said that you **must avoid using GlobalScope**. This is an antipattern, and something that you **shouldn't** use in ViewModels or Activities/Fragments. Use `viewModelScope` instead.

Reply:
I checked out another tutorial from RW, that suggests couroutineScope, is that ok? That works for me, but not sure if this is sufficient since you mentioned viewModelScope